zio-http/src/test/scala/zhttp/http/HttpDataSpec.scala

```diff
@@ -41,7 +62,88 @@ object HttpDataSpec extends DefaultRunnableSpec {
               assertM(res)(equalTo("abc\nfoo"))
             },
           ),
+          suite("UnsafeAsync")(
+            testM("finish on last message.") {
+              val unsafeChannel = new UnsafeReadableChannel {
+                override def read(): Unit = ()
+              }
+
+              val probe  = HttpData.UnsafeAsync { cb =>
+                val producer = cb(unsafeChannel)
+
+                producer(new UnsafeContent(new DefaultLastHttpContent(Unpooled.EMPTY_BUFFER)))
+              }
+              val result = probe.toByteBufStream(ByteBufConfig(4)).runCount
+              assertM(result)(equalTo(1L))
+            },
+            testM("wait for last message") {
+              val unsafeChannel = new UnsafeReadableChannel {
+                override def read(): Unit = ()
+              }
+
+              val probe  = HttpData.UnsafeAsync { cb =>
+                val producer = cb(unsafeChannel)
+                producer(
+                  new UnsafeContent(new DefaultHttpContent(Unpooled.copiedBuffer("abc".toArray.map(_.toByte)))),
+                )
+              }
+              val result = probe.toByteBufStream(ByteBufConfig(4)).runCount
+              assertM(result)(equalTo(1L))
+            } @@ timeout(1 seconds) @@ failing,
+            testM("process all messages.") {
```

Comment:
this and the next test are doing the same thing

Reply:
done